### ISSUE TITLE
GUI issue after sysupgrade from 15.05.1 snapshot to 18.06.8.

### ISSUE THREAD

**Issue by roshavagarga** [openwrt/luci#4323]
The device in question is a `TL-WR841N v9.1`. I was running an ancient snapshot somebody had made for me years ago and decided to do a sysupgrade to the newest tiny version with 'keep settings' checked.

When trying to access LuCi through 192.168.1.1 I get:
```
/usr/lib/lua/luci/dispatcher.lua:234: /etc/config/luci seems to be corrupt, unable to find section 'main'
stack traceback:
	[C]: in function 'assert'
	/usr/lib/lua/luci/dispatcher.lua:234: in function 'dispatch'
	/usr/lib/lua/luci/dispatcher.lua:127: in function </usr/lib/lua/luci/dispatcher.lua:126>
```

Now, I know this might be an issue with `rpcd` not running, but I am what is known as an utter noob. Running Windows 7, putty and winscp both gave me 'connection refused' issues when I tried to use those to connect to the router. I'd be grateful if somebody could point me to an ELI5 tutorial on how to get rpcd running properly.

I know the router is old, I'm basically using it for a 2.4ghz router, while my main router is only running 5ghz. I will note that despite the GUI not running, the network itself is running properly, just wanted to make sure I can fix it in case I want to tweak any settings.

**Comment by translucentfocus** [comment]
Thanks for posting your error message!  Just trusting the error message, it looks like you have "main" missing from your luci config file.

Your luci config should start out with something like this:
```
config core 'main'
      option lang 'auto'
      option resourcebase '/luci-static/resources'
      option mediaurlbase '/luci-static/material'
```

I _think_ your bigger problem is that you can't putty in to fix it... so I don't know how would do any of that without just resetting the router.  I'm willing to help, but if you can't ssh and can't access the gui... I don't know of a simple way to access the router configs.

**Comment by roshavagarga** [comment]
I might just be doing the putty thing wrong for all I know.

**Comment by translucentfocus** [comment]
> 
> 
> I might just be doing the putty thing wrong for all I know.

![image](https://user-images.githubusercontent.com/26986716/90568772-2c102b80-e172-11ea-8248-3596fce03a7c.png)

If it looks like that, you should be able to access it.  If it doesn't even open a console... I'm afraid you'll just have to reset.

[Keep Settings](https://openwrt.org/docs/guide-quick-start/admingui_sysupgrade_keepsettings) should not be used when upgrading as you did... so I'm surprised your network is operating at all 🙂

Specifically:

"Only use it for the same firmware channel (release → release, snapshot → snapshot)."
"Cautiousness is mostly needed with major version upgrades, e.g. from 15.05 to 17.01, or from 15.05 to 18.06 etc."

**Comment by feckert** [comment]
You have two option:
- Factory reset the router so you have a fresh installed system and reconfigure it again.
- Fix the luci config as @translucentfocus suggests and check if it works.
